frontend/src/bundles/chat/pages/chat-modal.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+    Modal,
+    ModalCloseButton,
+    ModalContent,
+    ModalOverlay,
+} from '~/bundles/common/components/components.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { Chat } from '../components/components.js';
+import { MessageSender } from '../enums/message-sender.js';
+import { actions as chatActions } from '../store/chat.js';
+import { type GenerateTextRequestDto, type Message } from '../types/types.js';
+
+type Properties = {
+    isChatOpen: boolean;
+    onModalChatClose: () => void;
+};
+
+const ChatModal: React.FC<Properties> = ({ isChatOpen, onModalChatClose }) => {
+    const dispatch = useAppDispatch();
+    const { messages } = useAppSelector(({ chat }) => ({
+        messages: chat.messages,
+    }));
+
+    const handleSendMessage = useCallback(
+        (payload: GenerateTextRequestDto) => {
+            const userMessage: Message = {
+                id: messages.length + 1,
+                sender: MessageSender.USER,
+                text: payload.message,
+                timeStamp: new Date(),
```

Comment:
Maybe, it would be better to add `createMessage` helper to create user and ai message?

Also it would be better to create message in slice reducer as you do it with ai messages